pkg/system/hub/remote/enroll.go

```diff
@@ -49,16 +49,20 @@ func Enroll(ctx context.Context, enrollment *gen.Enrollment, authority pki.Sourc
 
 	// the certInterceptor captures and saves the certificate presented by the server when the connection is opened
 	creds := &certInterceptor{TransportCredentials: credentials.NewTLS(tlsConfig)}
-	conn, err := grpc.DialContext(ctx, enrollment.TargetAddress,
+	conn, err := grpc.NewClient(enrollment.TargetAddress,
 		grpc.WithTransportCredentials(creds),
-		grpc.WithBlock(),
-		grpc.FailOnNonTempDialError(true),
-		grpc.WithReturnConnectionError(),
 	)
 	if err != nil {
 		return nil, err
 	}
 
+	client := gen.NewEnrollmentApiClient(conn)
+	// any api call will do to force the connection to be established (or fail)
```

Comment:
`WaitForReady` would achieve the same and behave more similarly to the previous implementation, no?

Reply:
Unfortunately there's no way to accurately replicate the old behaviour without using an RPC. The trouble is there's no public way to get the connection error that cause the conn status to enter TransientFailure. This means we can't distinguish between temporary and permanent errors, or even know what the error is.

I figured it would be safer to use the recommended approach (do an RPC) here instead of either masking the underlying error (replace it with client.ErrTransientFailure) or failing too early (because we can't recognise temp/perm errors).

Do you think I need to say that anywhere?

Reply:
No, I think it's fine